tests/components/energyzero/test_sensor.py

```diff
@@ -0,0 +1,140 @@
+"""Tests for the sensors provided by the EnergyZero integration."""
+from homeassistant.components.energyzero.const import DOMAIN
+from homeassistant.components.sensor import (
+    ATTR_STATE_CLASS,
+    SensorDeviceClass,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_FRIENDLY_NAME,
+    ATTR_ICON,
+    ATTR_UNIT_OF_MEASUREMENT,
+    CURRENCY_EURO,
+    ENERGY_KILO_WATT_HOUR,
+    VOLUME_CUBIC_METERS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from tests.common import MockConfigEntry
+
+
+async def test_energy_today(
+    hass: HomeAssistant, init_integration: MockConfigEntry
+) -> None:
+    """Test the EnergyZero - Energy sensors."""
+    entry_id = init_integration.entry_id
+    entity_registry = er.async_get(hass)
+    device_registry = dr.async_get(hass)
+
+    # Current energy price sensor
+    state = hass.states.get("sensor.energyzero_today_energy_current_hour_price")
+    entry = entity_registry.async_get(
+        "sensor.energyzero_today_energy_current_hour_price"
+    )
+    assert entry
+    assert state
+    assert entry.unique_id == f"{entry_id}_today_energy_current_hour_price"
+    assert state.state == "unknown"
```

Comment:
This value should not be none, but run into problems mocking what the current datetime is.

Reply:
Fixed with using `freeze_time`